vars/tdr.groovy

```diff
@@ -83,14 +102,18 @@ def createGitHubPullRequest(Map params) {
 }
 
 // This is used to get the URL needed to send a POST request to the GitHub API to update the specified repo with the Jenkins build status. This returns the API URL.
-def githubApiStatusUrl(String repo) {
-  String cmd = "git rev-parse HEAD"
-  if(sh(script: cmd, returnStatus: true) == 128) {
-    checkout scm
+def githubApiStatusUrl(String repo, String sha='null') {
```

Comment:
Could the the value be `null` rather than the string `'null'`? If that doesn't work, it would be better to give the string a more readable default value like `'UNKNOWN_SHA'`.

It's not a big deal here, in a temporary method parameter, but it's a good habit to get into because strings with the value `'null'` can cause a lot of confusion because they look like `null` at first glance but they're not actually null!